packages/next/src/server/app-render/app-render.tsx

```diff
@@ -2203,3 +2746,58 @@ async function prerenderToStream(
     }
   }
 }
+
+const loadingChunks: Set<Promise<unknown>> = new Set()
+const chunkListeners: Array<(x?: unknown) => void> = []
+
+function trackChunkLoading(load: Promise<unknown>) {
+  loadingChunks.add(load)
+  load.finally(() => {
+    if (loadingChunks.has(load)) {
+      loadingChunks.delete(load)
+      if (loadingChunks.size === 0) {
+        // We are not currently loading any chunks. We can notify all listeners
+        for (let i = 0; i < chunkListeners.length; i++) {
+          chunkListeners[i]()
+        }
+        chunkListeners.length = 0
+      }
+    }
+  })
+}
+
+export async function warmFlightResponse(
+  flightStream: BinaryStreamOf<any>,
+  clientReferenceManifest: DeepReadonly<ClientReferenceManifest>
+) {
+  let createFromReadableStream
+  if (process.env.TURBOPACK) {
+    createFromReadableStream =
+      // eslint-disable-next-line import/no-extraneous-dependencies
+      require('react-server-dom-turbopack/client.edge').createFromReadableStream
+  } else {
+    createFromReadableStream =
+      // eslint-disable-next-line import/no-extraneous-dependencies
+      require('react-server-dom-webpack/client.edge').createFromReadableStream
+  }
+
+  try {
+    createFromReadableStream(flightStream, {
+      ssrManifest: {
+        moduleLoading: clientReferenceManifest.moduleLoading,
+        moduleMap: clientReferenceManifest.ssrModuleMapping,
+      },
+    })
+  } catch (err) {
+    // We don't want to handle errors here but we don't want it to
+    // interrupt the outer flow. We simply ignore it here and expect
+    // it will bubble up during a render
```

Comment:
Shouldn't we then also await the `createFromReadableStream` call above (returning the root chunk)?

Reply:
I don't want to initialize any models because this response isn't the one that will be used by the SSR render. We know the response will be able to start loading chunks synchronously because we don't even get to the part where we warmup the response until we've completely finished rendering the RSC stream. Is there another reason to wait for the root chunk?